### DISCUSSION TITLE
How to put a context menu of actions on a MFXTableRow

### DISCUSSION THREAD

**Discussion by infinite-dev22** [palexdev/MaterialFX#320] [General]
I want to have some options presented to the user when he/she right clicks on a table row.
I tried this bellow:
  
    branchTable.setTableRowFactory(tableRow -> {  
		// Instanciate a row. Trying to get a row by this.  
		MFXTableRow<Branch> row = new MFXTableRow<>(); // wrong usage.  
  
		// Use the above row and place a MouseEvent Listener.  
		row.setOnMouseClicked(e -> {  
			if (e.getButton().equals(MouseButton.SECONDARY)) {  
				// Display Context Menu.  
			}  
		});  
	});

There isn't away for to place an event listener on the table rows.
Thanks in advance.

**Comment by palexdev** [comment]
I believe you are using it wrong. The table row factory is a function that maps T to a MFXTableRow. And sice it's a function you also need to return a value, so the code you posted won't even compile

```java
table.setTableRowFactory(t - > {
    MFXTableRow<T> row = new MFXTableRow<>(t);
    row.setOnMouseClicked(e -> ...);
    return row;
});
```
This should work

**Reply by infinite-dev22** [reply]
I get this:
![Screenshot from 2023-05-16 22-36-35](https://github.com/palexdev/MaterialFX/assets/77667007/a14e32c5-85d7-4573-8886-5893fe761c28)
won't compile.
Am using JDK 19.

**Reply by palexdev** [reply]
A yes the constructor also needs the table instance

**Reply by infinite-dev22** [reply]
Thanks, that gets it working 😀

**Reply by infinite-dev22** [reply]
I shall close this then.
